File: src/io/internal/extensionToImageIO.ts
const extensionToIO = new Map([
  ['bmp', 'itkBMPImageIO'],
  ['BMP', 'itkBMPImageIO'],

  ['dcm', 'itkGDCMImageIO'],
  ['DCM', 'itkGDCMImageIO'],

  ['gipl', 'itkGiplImageIO'],
  ['gipl.gz', 'itkGiplImageIO'],

  ['hdf5', 'itkHDF5ImageIO'],

  ['jpg', 'itkJPEGImageIO'],
  ['JPG', 'itkJPEGImageIO'],
  ['jpeg', 'itkJPEGImageIO'],
  ['JPEG', 'itkJPEGImageIO'],

  ['iwi', 'itkWASMImageIO'],
  ['iwi.cbor', 'itkWASMImageIO'],

  ['lsm', 'itkLSMImageIO'],

  ['mnc', 'itkMINCImageIO'],
  ['MNC', 'itkMINCImageIO'],
  ['mnc.gz', 'itkMINCImageIO'],
  ['MNC.GZ', 'itkMINCImageIO'],
  ['mnc2', 'itkMINCImageIO'],
  ['MNC2', 'itkMINCImageIO'],

  ['mgh', 'itkMGHImageIO'],
  ['mgz', 'itkMGHImageIO'],
  ['mgh.gz', 'itkMGHImageIO'],

  ['mha', 'itkMetaImageIO'],
  ['mhd', 'itkMetaImageIO'],

  ['mrc', 'itkMRCImageIO'],

  ['nia', 'itkNiftiImageIO'],
  ['nii', 'itkNiftiImageIO'],
  ['nii.gz', 'itkNiftiImageIO'],
  ['hdr', 'itkNiftiImageIO'],

  ['nrrd', 'itkNrrdImageIO'],
  ['NRRD', 'itkNrrdImageIO'],
  ['nhdr', 'itkNrrdImageIO'],
  ['NHDR', 'itkNrrdImageIO'],

  ['png', 'itkPNGImageIO'],
  ['PNG', 'itkPNGImageIO'],

  ['pic', 'itkBioRadImageIO'],
  ['PIC', 'itkBioRadImageIO'],

  ['tif', 'itkTIFFImageIO'],
  ['TIF', 'itkTIFFImageIO'],
  ['tiff', 'itkTIFFImageIO'],
  ['TIFF', 'itkTIFFImageIO'],

  ['vtk', 'itkVTKImageIO'],
  ['VTK', 'itkVTKImageIO'],

  ['isq', 'itkScancoImageIO'],
  ['ISQ', 'itkScancoImageIO'],

  ['fdf', 'itkFDFImageIO'],
  ['FDF', 'itkFDFImageIO']
])

export default extensionToIO

File: packages/image-io/python/itkwasm-image-io-emscripten/itkwasm_image_io_emscripten/extension_to_image_io.py
from collections import OrderedDict

extension_to_image_io = OrderedDict([
  ('.bmp', 'bmp'),

  ('.dcm', 'gdcm'),

  ('.gipl', 'gipl'),
  ('.gipl.gz', 'gipl'),

  ('.hdf5', 'hdf5'),

  ('.jpg', 'jpeg'),
  ('.jpeg', 'jpeg'),

  ('.iwi', 'wasm'),
  ('.iwi.cbor', 'wasm'),
  ('.iwi.cbor.zst', 'wasmZstd'),

  ('.lsm', 'lsm'),

  ('.mnc', 'mnc'),
  ('.mnc.gz', 'mnc'),
  ('.mnc2', 'mnc'),

  ('.mgh', 'mgh'),
  ('.mgz', 'mgh'),
  ('.mgh.gz', 'mgh'),

  ('.mha', 'meta'),
  ('.mhd', 'meta'),

  ('.mrc', 'mrc'),

  ('.nia', 'nifti'),
  ('.nii', 'nifti'),
  ('.nii.gz', 'nifti'),
  ('.hdr', 'nifti'),

  ('.nrrd', 'nrrd'),
  ('.nhdr', 'nrrd'),

  ('.png', 'png'),

  ('.pic', 'bioRad'),

  ('.tif', 'tiff'),
  ('.tiff', 'tiff'),

  ('.vtk', 'vtk'),

  ('.isq', 'scanco'),
  ('.aim', 'scanco'),

  ('.fdf', 'fdf'),
])
